test/validator/validation-result.test.ts

```diff
@@ -0,0 +1,37 @@
+import {ValidationResult} from '../../src/validator/validation-result';
+
+describe('ValidationResult', () => {
+  it('is valid by default', () => {
+    const result = new ValidationResult();
+    expect(result.isValid()).toBe(true);
+  });
+
+  it('can be created with a static valid method', () => {
+    const result = ValidationResult.valid();
+    expect(result.isValid()).toBe(true);
+  });
+
+  it('is invalid if an error has been logged', () => {
+    const result = new ValidationResult();
+    result.putError('some kind of error happened');
```

Comment:
Would be nicer to test any 'known' issues/errors here. I'd also update the error handling in the code to handle specific errors in different ways to give the user better feedback.

Reply:
Afraid I'm in the `fail first, fail fast` camp on this one. I'd prefer either real scenarios and real errors or remove all of these tests. Currently, they only really test the built-in Array methods.

Reply:
My thinking on this is we should stick a comment into the test file - pointing out that they only test the array methods - and so the tests are only there as a sanity check.